mmeval/metrics/gradient_error.py

```diff
@@ -0,0 +1,192 @@
+        >>> pred_alpha = np.random.randn(32, 32).astype('uint8')
+        >>> gt_alpha = np.ones((32, 32), dtype=np.uint8) * 255
+        >>> trimap = np.zeros((32, 32), dtype=np.uint8)
+        >>> trimap[:16, :16] = 128
+        >>> trimap[16:, 16:] = 255
+        >>> gradient_error(pred_alpha, gt_alpha, trimap)  # doctest: +ELLIPSIS
+        {'GradientError': ...}
+    """
+
+    def __init__(self,
+                 sigma: float = 1.4,
+                 norm_const: int = 1000,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.sigma = sigma
+        self.norm_const = norm_const
+
+        if cv2 is None:
+            raise ImportError(f'For availability of {self.__class__.__name__},'
+                              ' please install cv2 first.')
+
+    def add(self, pred_alphas: Sequence[np.ndarray], gt_alphas: Sequence[np.ndarray], trimaps: Sequence[np.ndarray]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add GradientError score of batch to ``self._results``
+
+        Args:
+            pred_alphas (Sequence[np.ndarray]): Pred_alpha data of predictions.
+            ori_alphas (Sequence[np.ndarray]): Ori_alpha data of data_batch.
+            ori_trimaps (Sequence[np.ndarray]): Ori_trimap data of data_batch.
+        """
+
+        for pred_alpha, gt_alpha, trimap in zip(pred_alphas, gt_alphas,
+                                                trimaps):
+            assert pred_alpha.shape == gt_alpha.shape, 'The shape of ' \
+                '`pred_alpha` and `gt_alpha` should be the same, but got: ' \
+                f'{pred_alpha.shape} and {gt_alpha.shape}'
+
+            gt_alpha_normed = np.zeros_like(gt_alpha)
+            pred_alpha_normed = np.zeros_like(pred_alpha)
+
+            cv2.normalize(gt_alpha, gt_alpha_normed, 1.0, 0.0, cv2.NORM_MINMAX)
+            cv2.normalize(pred_alpha, pred_alpha_normed, 1.0, 0.0,
+                          cv2.NORM_MINMAX)
+
+            gt_alpha_grad = gauss_gradient(gt_alpha_normed, self.sigma)
+            pred_alpha_grad = gauss_gradient(pred_alpha_normed, self.sigma)
+            # this is the sum over n samples
+            grad_loss = ((gt_alpha_grad - pred_alpha_grad)**2 *
+                         (trimap == 128)).sum()
+
+            # divide by self.norm_const to reduce the magnitude of the result
+            grad_loss /= self.norm_const
+
+            self._results.append(grad_loss)
+
+    def compute_metric(self, results: List) -> Dict[str, float]:
+        """Compute the GradientError metric.
+
+        Args:
+            results (List): A list that consisting the GradientError score.
+                This list has already been synced across all ranks.
+
+        Returns:
+            Dict[str, float]: The computed GradientError metric.
+            The keys are the names of the metrics,
+            and the values are corresponding results.
+        """
+
+        return {'GradientError': float(np.array(results).mean())}
```

Comment:
Should `'GradientError'` be lower letters to make it as same as other metrics? @ice-tong

Reply:
Good point! We should unify the key style in the returned metric results. cc @xuan07472

Reply:
okay